src/PockyBot.NET/Services/Pegs/PegHelper.cs

```diff
@@ -45,5 +45,10 @@ public int GetPegWeighting(string senderLocation, string receiverLocation)
 
             return _configRepository.GetGeneralConfig("remoteLocationWeightingDefault") ?? DefaultRemoteLocationWeighting;
         }
+
+        public bool IsPenaltyPeg(string comment, int? requireKeywords, string[] penaltyKeywords)
+        {
+            return penaltyKeywords.Any(x => comment.IndexOf(x, StringComparison.OrdinalIgnoreCase) >= 0);
```

Comment:
a penalty peg is only a penalty if it also doesn't contain any valid keywords. i.e if it were to have both "shame" and "amazing" it would still count as a valid peg

Reply:
Fixed.